hal/network/ncp_client/sara/sara_ncp_client.cpp

```diff
@@ -2182,14 +2228,22 @@ int SaraNcpClient::getMtu() {
 int SaraNcpClient::urcs(bool enable) {
     const NcpClientLock lock(this);
     if (enable) {
+        sleepNoPPPWrite_ = false;
         CHECK_TRUE(muxer_.resumeChannel(UBLOX_NCP_AT_CHANNEL) == 0, SYSTEM_ERROR_INTERNAL);
         if (ncpId() == PLATFORM_NCP_SARA_U201) {
             // Make sure the modem is responsive again. U201 modems do take a while to
             // go back into functioning state
             CHECK(waitAtResponse(5000, gsm0710::proto::DEFAULT_T2));
         }
     } else {
+        sleepNoPPPWrite_ = true;
+        // R510 may be in low power mode, wake it up so we can get our muxer data channel suspend
+        // echo, before we get into sleep and potentially prematurely wake by "network activity" (RX data)
+        if (ncpId() == PLATFORM_NCP_SARA_R510) {
+            CHECK(waitAtResponse(5000, 2000));
+        }
         CHECK_TRUE(muxer_.suspendChannel(UBLOX_NCP_AT_CHANNEL) == 0, SYSTEM_ERROR_INTERNAL);
+        HAL_Delay_Milliseconds(100); // allow a bit of time for muxer echo response
```

Comment:
Should also set keepalive to 0 here perhaps?

Reply:
I'm guessing you mean the PPP echo timer, since the muxer AT channel is already suspended?  PPP data writes should also be blocked with `sleepNoPPPWrite_` already?

Reply:
No, I was mainly thinking about muxer keepalives, to avoid sending a keepalive right before sleep, going into it and then waking up wrongly.

Reply:
I tested `muxer_.suspendChannel()` and it in fact does not stop muxer keepAlives on the AT channel.  So even for R410, this would prevent it from staying asleep while trying to wake on network activity.  I added some code to force this to be disabled when we disable urcs (presumed for sleep usage).